src/main/java/emissary/output/DropOffUtil.java

```diff
@@ -969,32 +983,63 @@ public Date extractEventDateFrom(final IBaseDataObject d, final boolean lastReso
-     * IBaseDataObject's "FILEXT" parameter is set as the unique set of extracted file extensions, converted to lowercase.
+     * Utilizes the static methods getFullFilepathsFromParams and getFileExtensions to extract the file extensions from all
+     * the filenames of the object of a given {@link IBaseDataObject}. If one or more file extensions are extracted, the
+     * IBaseDataObject's FILEXT parameter is set as the unique set of extracted file extensions, converted to lowercase.
      *
      * @param p IBaseDataObject to process
      *
      */
     void extractUniqueFileExtensions(IBaseDataObject p) {
-        if (p.hasParameter(ORIGINAL_FILENAME)) {
-            final Set<String> extensions = new HashSet<>();
-            for (Object filename : p.getParameter(ORIGINAL_FILENAME)) {
-                final String fn = (String) filename;
-                if (StringUtils.isNotEmpty(fn) && fn.lastIndexOf('.') > -1) {
-                    final int pos = fn.lastIndexOf('.') + 1;
-                    if (pos < fn.length()) {
-                        final String fext = fn.substring(pos).toLowerCase();
-                        if (fext.length() > 0 && fext.length() <= this.maxFilextLen) {
-                            extensions.add(fext);
-                        }
+        List<String> filenames = getFullFilepathsFromParams(p);
+        Set<String> extensions = getFileExtensions(filenames, this.maxFilextLen);
+        if (!extensions.isEmpty()) {
+            p.setParameter(FILEXT, extensions);
+        }
+    }
+
+    /**
+     * Given a list of filenames, extract and return a set of non-blank file extensions converted to lowercase.
+     *
+     * @param filenames The list of filenames to examine
+     * @param maxFilextLen The maximum size we want a file extension to be
+     * @return A set of unique file extensions from the filename list
+     */
+    public static Set<String> getFileExtensions(List<String> filenames, int maxFilextLen) {
+        final Set<String> extensions = new HashSet<>();
+        for (String filename : filenames) {
+
+            // add the file extension if it is smaller than maxFileextLen
+            final String fext = FilenameUtils.getExtension(filename);
+            if (StringUtils.isNotBlank(fext) && fext.length() <= maxFilextLen) {
+                extensions.add(fext.toLowerCase());
+            }
+        }
+        return extensions;
+    }
+
+    /**
+     * Checks the Original-Filename and FILE_ABSOLUTEPATH for the filename of the object. Returns a list with of the
+     * non-empty strings found in these fields. If nothing is found in either field, return an empty list.
+     *
+     * @param d The IBDO
+     * @return The list of filenames found in the field Original-Filename or FILE_ABSOLUTEPATH
+     */
+    public static List<String> getFullFilepathsFromParams(IBaseDataObject d) {
```

Comment:
Potential race condition for field`defaultFilenameFields` if the constructor `public DropOffUtil(final Configurator configG)` is called with a different set of configs.

Reply:
I added using a lock around the array initialization in configure and when we access it here which should ensure we aren't reading it and recreating it at the same time.

Reply:
Adding the lock doesn't address mutability issue of `defaultFilenameFields`, which can be change if a new instance of `DropOffUtil` is instantiated with a different config.

Here is a test explaining the issue:
`
    @Test
    void testModifiedFileNameFields() throws IOException {
        IBaseDataObject ibdo = new BaseDataObject();
        List<String> bestFilenames;

        ibdo.setParameter(ORIGINAL_FILENAME, "");
        ibdo.setParameter(FILE_ABSOLUTEPATH, "");
        bestFilenames = DropOffUtil.getFullFilepathsFromParams(ibdo);
        assertEquals(0, bestFilenames.size(), "No filename should have been found");

        ibdo.setParameter(FILE_ABSOLUTEPATH, "theOtherFile.csv");
        bestFilenames = DropOffUtil.getFullFilepathsFromParams(ibdo);
        assertEquals(1, bestFilenames.size(), "There should be one filename extracted");
        assertEquals("theOtherFile.csv", bestFilenames.get(0), "The FILE_ABSOLUTEPATH should have been extracted");

        ibdo.setParameter(ORIGINAL_FILENAME, "file.docx");
        bestFilenames = DropOffUtil.getFullFilepathsFromParams(ibdo);
        assertEquals(2, bestFilenames.size(), "There should be two filenames extracted");
        assertEquals("file.docx", bestFilenames.get(0), "The Original-Filename should have been extracted");
        assertEquals("theOtherFile.csv", bestFilenames.get(1), "The Original-Filename should have been extracted");

        Configurator config = ConfigUtil.getConfigInfo(DropOffUtil.class);
        config.removeEntry("FILENAME_FIELDS", "*");
        // Adding a field name that is not currently stored in the IBDO
        config.addEntry("FILENAME_FIELDS", "DIFFERENT_FILENAME_FIELD");

        // Contruction of DropOffUtil with a different config will affect the behavior of DropOffUtil.getFullFilepathsFromParams
        new DropOffUtil(config);

        bestFilenames = DropOffUtil.getFullFilepathsFromParams(ibdo);
        // Expect this test to now fail with the updated FILENAME_FIELDS configs
        assertEquals(2, bestFilenames.size(), "There should be two filenames extracted");

    }
    ```

Reply:
Thanks for that example. I'm thinking to create a new util class for the two static functions to avoid the issue of multiple DropOffUtil instances being created with different configs.

Reply:
I resolved this issue by removing the ability to specify these fields in a config file. I overloaded the function getFullFilepathsFromParams so we can optionally provide a list of fields to look in instead of the default fields, in case there is a situation where we need this flexibility.